src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,52 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<byte> inputValue = value;
+
+            // We will ignore the leading and trailing occurrences of the white space in the json check.
+            int start = 0;
+            while (start < inputValue.Length && char.IsWhiteSpace(Convert.ToChar(inputValue[start])))
+            {
+                start++;
+            }
+            int end = inputValue.Length - 1;
+            while (end >= start && char.IsWhiteSpace(Convert.ToChar(inputValue[end])))
+            {
+                end--;
+            }
+            int len = end - start + 1;
+
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json string, Json value true, Json value false, Json number.
+            // Our logic here is that, for example, we check if the first and the last bytes are '{' and '}', respectively.
```

Comment:
Is this heuristic enough? What about values like `{   this{ is { invalid { josn }` ?

Reply:
Taking a step back, is it even valid to store use `BinaryData` to store values that cannot be serialized to json?

Reply:
oh -  I see now that we handle the edge case in the catch block

Reply:
Yes, we think that this check will be right in most of the cases and use catch to handle the worst case, in which the `BinaryData` passed the check but isn't a valid json. We use this fast check to avoid trying to parse the json everytime and then catch the exception.